src/flamegpu/util/detail/JitifyCache.cu

```diff
@@ -315,12 +317,20 @@ std::unique_ptr<KernelInstantiation> JitifyCache::compileKernel(const std::strin
     }
 #endif
 
-    // Set the compilation architecture target if it was successfully detected.
-    int currentDeviceIdx = 0;
-    cudaError_t status = cudaGetDevice(&currentDeviceIdx);
-    if (status == cudaSuccess) {
-        int arch = compute_capability::getComputeCapability(currentDeviceIdx);
-        options.push_back(std::string("--gpu-architecture=compute_" + std::to_string(arch)));
+    // Set the cuda compuate capability architecture to optimize / generate for, based on the values supported by the current dynamiclaly linked nvrtc and the device in question.
+    std::vector<int> nvrtcArchitectures = util::detail::compute_capability::getNVRTCSupportedComputeCapabilties();
+    if (nvrtcArchitectures.size()) {
+        int currentDeviceIdx = 0;
+        if (cudaSuccess == cudaGetDevice(&currentDeviceIdx)) {
+            int arch = compute_capability::getComputeCapability(currentDeviceIdx);
+            int maxSupportedArch = compute_capability::selectAppropraiteComputeCapability(arch, nvrtcArchitectures);
+            // only set a nvrtc compilation flag if a usable value was found
+            if (maxSupportedArch != 0) {
+                options.push_back(std::string("--gpu-architecture=compute_" + std::to_string(maxSupportedArch)));
+            }
+            // Optionally we could error here, but this *should* never occur
```

Comment:
That's what an `assert(false)` or `InternalError` is for. Adding the check would be harmless additional safety.

Reply:
Added the `assert(false)`, users wouldn't be able to catch the exception so continuing with generic arch RTC seems like the better choice.